packages/webapp/src/containers/hooks/useSearchFilter.js

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState, useMemo } from 'react';
+
+/**
+ * A custom hook to filter an array of items based on a string.
+ *
+ * This hook can be used with either the search string controlled in state, or set elsewhere and passed in. If a search string is not provided, internal state will be used.
+ *
+ * @param {Array} items - The items to filter.
+ * @param {function} getSearchableString - Function that takes in a array item and constructs the string to search on (e.g. concatenates the properties of interest)
+ * @returns {Object} - An object containing the current filter, a setter for the filter, and the filtered items.
+ */
+
+export const useSearchFilter = (
+  items = [],
+  getSearchableString = () => {},
+  providedFilter = '',
+) => {
+  const [internalFilter, setInternalFilter] = useState('');
+
+  const searchString = providedFilter || internalFilter;
```

Comment:
I think it's a bit confusing here that if `providedFilter`, it will cause `internalFilter